### PR TITLE
fix: Fix index out of bounds error in nomos status command

### PR DESCRIPTION
**Pull request by rquitales**
This PR fixes nomos status to prevent it from panicing by ensuring the slice it references is of the right length.

### CODE REVIEW

File: cmd/nomos/status/client.go
Comment (janetkuo) [inline cmd/nomos/status/client.go:287]: This is implementation detail in `namespaceRGs` creation and not guaranteed to be always true (someone might change the function and break it in the future). Check that the length are identical and append an error if not, so that we won't hit panic errors.

Comment (rquitales) [reply cmd/nomos/status/client.go:287]: Good idea. Added a check to also ensure both slices are of the same length.

File: cmd/nomos/status/client.go
Comment (nan-yu) [inline cmd/nomos/status/client.go:266]: Can you add namespace to the error so that we know if that's for root-sync or repo-sync, and all other resourcegroups as well?

Comment (rquitales) [reply cmd/nomos/status/client.go]: Done.

**Review by nan-yu — APPROVED** [review]
Do you want to merge the change into the main branch? 

I added a `/hold` comment to hold the merge.  Feel free to `/unhold` to get it into the current release.

/lgtm
/hold

File: cmd/nomos/status/client.go
Comment (karlkfi) [inline cmd/nomos/status/client.go:265]: I imagine it's possible that there's no RG if there was an error before it was created. Will this suppress or expose that kind of error?

Comment (rquitales) [reply cmd/nomos/status/client.go:265]: If there is no RG because of an error before it was created, we should still report that the RG is missing in `nomos status` right? We should always have a ResourceGroup per R*Sync I belive.

Comment (rquitales) [reply cmd/nomos/status/client.go:265]: Previously when nomos CLI used a GET method, we'd always surface errors when we can't find the ResourceGroup. A regression occurred when we updated the status command to LIST all ResourceGroups, thus we didn't surface that the ResourceGroup is missing after that change was merged.

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**
